ducktape/tests/regression_test.py

```diff
@@ -0,0 +1,64 @@
+# Copyright 2015 Confluent Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from ducktape.tests.test import Test, TestContext
+from ducktape.tests.result_store import TestKey
+from ducktape.mark._parametrize import _inject
+
+import json
+import statistics
+
+
+class RegressionTest(Test):
+    pass
+
+
+class DefaultRegressionTest(RegressionTest):
+    """
+    Only run if some test with matching TestKey gets run
```

Comment:
This could use more detail about what the default regression test actually does, i.e. when will it fail, when will it succeed?